Fix ReAct agent TypeError with LiteLLM and Anthropic models

sjarmak opened the pull request:
## Description
Fixes the ReAct agent initialization failure when using LiteLLM provider with Anthropic models. The agent was throwing `TypeError: expected string or bytes-like object, got 'NoneType'` during model binding.

## Changes
- Modified `_maybe_bind_llm_and_yield()` method in `src/nemo_agent/react_agent/agent.py` (line 125)
- Removed redundant check for `model_name` attribute (always `None` in `ChatLiteLLM`)
- Check only the `model` attribute and add defensive `str()` conversion
- Works across all LiteLLM providers (OpenAI, Azure, Anthropic, Bedrock, Cohere, Groq)

## Root Cause
When using `ChatLiteLLM` (LangChain's wrapper for LiteLLM), the `model_name` attribute is always `None`. The code attempted to regex search on both `model` and `model_name`, causing a TypeError when `model_name` was `None`.

## Testing
- Verified `model_name` is always `None` across multiple providers
- Confirmed `model` attribute contains the model string for all providers  
- Tested that regex correctly identifies GPT-5 models
- Patch allows ReAct agent to initialize with Anthropic models without errors
- MCP tools load and agent makes tool calls successfully

Closes #1250

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Chores**
  * Improved LLM model detection logic for more reliable configuration handling across different model types.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

willkill07 reviewed (changes requested):
This PR currently deletes most of the code from the codebase. Please ensure that only changes to the necessary files are made.

sjarmak committed:
Fix ReAct agent LiteLLM TypeError with Anthropic models

Fixes the ReAct agent initialization failure when using LiteLLM provider
with Anthropic models. The agent was throwing TypeError: expected string
or bytes-like object, got 'NoneType' during model binding.

Changes:
- Modified _maybe_bind_llm_and_yield() method in react_agent/agent.py
- Removed redundant check for model_name attribute (always None in ChatLiteLLM)
- Check only the model attribute with defensive str() conversion
- Works across all LiteLLM providers (OpenAI, Azure, Anthropic, Bedrock, etc.)

Root Cause:
When using ChatLiteLLM (LangChain's wrapper for LiteLLM), the model_name
attribute is always None. The code attempted to regex search on both model
and model_name, causing a TypeError when model_name was None.

Signed-off-by: Stephanie Jarmak <steph.jarmak@gmail.com>

willkill07 commented on src/nat/agent/react_agent/agent.py:
```diff
@@ -122,7 +122,7 @@ def _maybe_bind_llm_and_yield(self) -> Runnable[LanguageModelInput, BaseMessage]
         # models that don't need (or don't support)a stop sequence
         smart_models = re.compile(r"gpt-?5", re.IGNORECASE)
-        if any(smart_models.search(getattr(self.llm, model, "")) for model in ["model", "model_name"]):
+        if smart_models.search(str(getattr(self.llm, "model", ""))):
```
You are generalizing a bit too much here... `model_name` is still important to have:

```suggestion
        if any(smart_models.search(str(getattr(self.llm, model, ""))) for model in ["model", "model_name"]):
```

Wrapping the call to `getattr` with `str()` (as you suggested) should work.

willkill07 commented:
/ok to test [ee9fe3d](https://github.com/NVIDIA/NeMo-Agent-Toolkit/pull/1251/commits/ee9fe3d56cc70b8ff5c0c9acb66690e701b3c246)

sjarmak commented:
> This PR currently deletes most of the code from the codebase. Please ensure that only changes to the necessary files are made.

Oof, sorry, when I pulled the codebase down it must have been a shallow clone initially / not synced with upstream.

sjarmak committed:
Merge branch 'develop' into sjarmak/branch-1

willkill07 commented:
/ok to test [e923841](https://github.com/NVIDIA/NeMo-Agent-Toolkit/pull/1251/commits/e9238412717019b046c2d8e62ce9224e3c0cbbac)

willkill07 commented:
/merge

willkill07 commented:
@sjarmak thank you for your contribution!
